main.go

```diff
@@ -10,22 +10,28 @@ import (
 	"github.com/okkur/reposeed/cmd/reposeed/config"
 )
 
+const SupportedConfigVersion = "v1"
+
 func main() {
 	err := godotenv.Load()
 	if err != nil {
 		log.Println("Couldn't find .env file. Reading environment variables from system")
 	}
 	app := gin.Default()
 	config := &config.Config{}
-	app.POST("/generate", func(ctx *gin.Context) {
-		ctx.BindJSON(config)
-		filename, err := generator.CreateFiles(*config, config.Project.Name, os.Getenv("STORAGE"))
-		if err.Code != 200 {
-			ctx.JSON(400, err)
-			ctx.Abort()
-		} else {
-			ctx.File(filename)
-		}
-	})
-	app.Run(os.Getenv("PORT"))
+	if config.Project.Version == SupportedConfigVersion {
+		app.POST("/generate", func(ctx *gin.Context) {
+			ctx.BindJSON(config)
+			filename, err := generator.CreateFiles(*config, config.Project.Name, os.Getenv("STORAGE"))
+			if err.Code != 200 {
+				ctx.JSON(400, err)
+				ctx.Abort()
+			} else {
+				ctx.File(filename)
+			}
+		})
+		app.Run(os.Getenv("PORT"))
+	} else {
+		log.Fatalf("Invalid config version. Currently supported versions: %s", SupportedConfigVersion)
```

Comment:
Because we are working on an HTTP API, we should return an HTTP response to the user. Return a 422 response containing the error message. In JSON format. Like this:
```json
{
  "error": {
    "ConfigVersion": "Invalid config version"
  }
}
```

Reply:
I noticed that I've written another format for the error response. we have 2 options:
* Rewrite the current JSONerror struct to make it like the example above (I personally prefer this one)
* Return this error like the previous format using the JSONerror struct.

The first option is better because it's a standard error response format in designing APIs.

Reply:
Ok, I've changed the JSONError struct and provided a 422 error for an invalid config version instead of the log.Fatalf(). Happy to make additional changes if I missed anything or need to do things differently.